src/entity/projectile/WindCharge.php

```diff
@@ -0,0 +1,242 @@
+			$impact = (1 - $distance) * $exposure;
+
+			if ($impact <= 0) continue;
+
+			if (round($entityPos->getX(), 1) == round($source->getX(), 1) && round($entityPos->getZ(), 1) == round($source->getZ(), 1)) {
+				$entity->setMotion($entity->getMotion()->add(0, 0.5 * $exposure, 0));
+
+				return;
+			}
+
+			$entity->setMotion($entity->getMotion()->add(0, $impact * 0.4, 0)->addVector($motion->multiply($impact)));
+		}
+	}
+
+	private function getBound(Vector3 $source, float $radius) : AxisAlignedBB {
+		$minX = (int) floor($source->x - $radius - 1);
+		$maxX = (int) ceil($source->x + $radius + 1);
+		$minY = (int) floor($source->y - $radius - 1);
+		$maxY = (int) ceil($source->y + $radius + 1);
+		$minZ = (int) floor($source->z - $radius - 1);
+		$maxZ = (int) ceil($source->z + $radius + 1);
+
+		return new AxisAlignedBB($minX, $minY, $minZ, $maxX, $maxY, $maxZ);
+	}
+
+	private function checkAffect(Block $affectedBlock) : void {
+		if($affectedBlock instanceof Trapdoor) {
+			/** @var Trapdoor $affectedBlock */
+
+			if($affectedBlock->getTypeId() == BlockTypeIds::IRON_TRAPDOOR) return;
+			$affectedBlock->setOpen(!$affectedBlock->isOpen());
+			$world = $affectedBlock->getPosition()->getWorld();
+			$world->setBlock($affectedBlock->getPosition(), $affectedBlock);
+			$world->addSound($affectedBlock->getPosition(), new DoorSound());
+
+			return;
+		}
+
+		if($affectedBlock instanceof Door) {
+			/** @var Door $affectedBlock */
+
+			if($affectedBlock->getTypeId() == BlockTypeIds::IRON_DOOR) return;
+			$affectedBlock->setOpen(!$affectedBlock->isOpen());
+			$world = $affectedBlock->getPosition()->getWorld();
+			$world->setBlock($affectedBlock->getPosition(), $affectedBlock);
+			$world->addSound($affectedBlock->getPosition(), new DoorSound());
+
+			return;
+		}
+
+		if($affectedBlock instanceof Button) {
+			/** @var Button $affectedBlock */
+
+			($affectedBlock->getTypeId() == BlockTypeIds::STONE_BUTTON || $affectedBlock->getTypeId() == BlockTypeIds::POLISHED_BLACKSTONE_BUTTON) ? $delay = 20 : $delay = 30;
+
+			$affectedBlock->setPressed(!$affectedBlock->isPressed());
+			$world = $affectedBlock->getPosition()->getWorld();
+			$world->setBlock($affectedBlock->getPosition(), $affectedBlock);
+			$world->scheduleDelayedBlockUpdate($affectedBlock->getPosition(), $delay);
+			$world->addSound($affectedBlock->getPosition()->add(0.5, 0.5, 0.5), new RedstonePowerOnSound());
+
+			return;
+		}
+
+		if($affectedBlock instanceof Lever) {
+			/** @var Lever $affectedBlock */
+
+			$affectedBlock->setActivated(!$affectedBlock->isActivated());
+			$world = $affectedBlock->getPosition()->getWorld();
+			$world->setBlock($affectedBlock->getPosition(), $affectedBlock);
+			$world->addSound(
+				$affectedBlock->getPosition()->add(0.5, 0.5, 0.5),
+				$affectedBlock->isActivated() ? new RedstonePowerOnSound() : new RedstonePowerOffSound()
+			);
+
+			return;
+		}
+
+		if($affectedBlock instanceof Bell) {
+			/** @var Bell $affectedBlock */
```

Comment:
Is this redundant? Most static analysis tools can infer this from the `instanceof` line.

Reply:
yes very much so